src/client.ts

```diff
@@ -230,12 +233,9 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
 
     this.userPromise = new Promise(resolve => {
       this.userPromiseResolver = resolve;
-    }).then(isUserResolved => {
-      if (isUserResolved) {
-        this.isUserReady = true;
-        return { success: true };
-      }
-      return { success: false, message: 'Failed to fetch qualified segments.' };
+    }).then((result: ResolveResult) => {
+      this.isUserReady = result.success;
+      return result;
     });
```

Comment:
I see resolver is set and called later. Wondering the value of this promise setting instead of calling line 237-238?